path_roles.go

```diff
@@ -166,6 +187,18 @@ func (b *azureSecretBackend) pathRoleUpdate(ctx context.Context, req *logical.Re
 		role.AzureRoles = parsedRoles
 	}
 
+	// Parse the Azure groups
+	if groups, ok := d.GetOk("azure_groups"); ok {
+		parsedGroups := make([]*AzureGroup, 0) // non-nil to avoid a "missing groups" error later
+
+		err := jsonutil.DecodeJSON([]byte(groups.(string)), &parsedGroups)
+		if err != nil {
+			return logical.ErrorResponse(fmt.Sprintf("error parsing Azure groups '%s': %s", groups.(string), err.Error())), nil
```

Comment:
```suggestion
			return logical.ErrorResponse("error parsing Azure groups '%s': %s", groups.(string), err.Error()), nil
```

Applies elsewhere too. Syntactical sugar now exists for this common pattern.